src/services/MailService.ts

```diff
@@ -1,53 +1,94 @@
-					filename,
-					content: Buffer.from(content, 'base64'),
-				}));
-			mail.attachments = decodeFiles(mail.attachments);
-		}
+export default class MailService {
+	private queueManager: QueueManager;
+	private platformIds: string[];
+	private transports: MailTransport[] = [];
 
-		const config = await Utils.getPlatformConfig(transporter.platformId);
-		if (config.mail.defaults.envelope) {
-			mail.envelope = {
-				from: config.mail.defaults.envelope.from || mail.from,
-				to: config.mail.defaults.envelope.to || mail.to,
-			};
-		}
+	/**
+	 *
+	 * @param queueManager
+	 * @param configuration
+	 */
+	constructor(queueManager: QueueManager, configuration: ConfigData[]) {
+		this.queueManager = queueManager;
+		this.platformIds = configuration.map((cfg) => {
+			queueManager.createQueue(SERVICE_TYPE, cfg.platformId, cfg.queue);
+
+			const mailOptions = Array.isArray(cfg.mail.options)
+				? (cfg.mail.options as ConfigData[])
+				: ([cfg.mail.options] as ConfigData[]);
+			mailOptions.forEach((options) => {
+				this.transports.push(new MailTransport(SERVICE_TYPE, cfg.platformId, { options, defaults: cfg.mail.defaults }));
+			});
 
-		return transporter.transporter.sendMail(mail);
+			return cfg.platformId;
+		});
 	}
 
-	protected _createTransporter(options: any, defaults: any): nodeMailer.Transporter {
-		// todo check default from becomes defined
-		return nodeMailer.createTransport(options, defaults);
+	/**
+	 *
+	 */
+	async startWorkers(): Promise<void> {
+		await Promise.all(
+			this.platformIds.map(async (platformId) => {
+				return this.queueManager.startWorker(SERVICE_TYPE, platformId, this.process.bind(this));
+			})
+		);
 	}
 
-	protected _createTransporters(platformId: string, config: any): PlatformMailTransporter[] {
-		const options = Array.isArray(config.mail.options) ? config.mail.options : [config.mail.options];
-		const defaults = config.mail.defaults;
-		const transporters = options.map((option: any) => this._createTransporter(option, defaults));
-		return transporters.map((transporter: nodeMailer.Transporter) => {
-			return {
-				platformId,
-				transporter,
-			};
-		});
+	/**
+	 *
+	 * @param platformId
+	 * @param message
+	 * @param receiver
+	 * @param messageId
+	 */
+	async send(platformId: string, message: PlatformMessage, receiver: string, messageId?: string): Promise<void> {
+		await this.queueManager.addJob({ serviceType: SERVICE_TYPE, platformId, message, receiver, messageId });
+		// TODO logging?
+		logger.debug('[queue] job added');
+	}
+
+	/**
+	 *
+	 * @param platformId
+	 * @param message
+	 * @param receiver
+	 * @param messageId
+	 */
+	async directSend(platformId: string, message: PlatformMessage, receiver: string, messageId?: string): Promise<void> {
+		logger.debug('direct send', { platformId, message, receiver, messageId });
+		await this.getTransport(SERVICE_TYPE, platformId).deliver(message as Mail);
+	}
+
+	private async process(job: Job<JobData>): Promise<void> {
```

Comment:
why not removing async and return this.getTransporter...deliver()? is'nt there a message id responded?

Reply:
I removed the unnecessary async/awaits.